File: rs/cyphrpass/src/transaction.rs
use crate::parsed_coz::VerifiedCoz;

/// A sequence of one or more related mutation cozies.
/// In Cyphrpass, mutation cozies MUST be grouped by transaction.
#[derive(Debug, Clone)]
pub struct Transaction(pub Vec<VerifiedCoz>);

/// The commit transaction is a specialized sequence of cozies containing the finality marker.
/// The commit transaction MUST be the exact last transaction in the commit.
#[derive(Debug, Clone)]
pub struct CommitTransaction(pub Vec<VerifiedCoz>);

File: go/cyphrpass/transaction.go
package cyphrpass

// Transaction is a sequence of related mutation cozies.
// In Cyphrpass, mutation cozies MUST be grouped by transaction.
type Transaction []*ParsedCoz

// CommitTransaction is a specialized sequence of cozies containing the finality marker.
// The commit transaction MUST be the exact last transaction in the commit.
type CommitTransaction []*ParsedCoz
